src/FoKL/FoKLRoutines.py

```diff
@@ -1695,115 +1698,120 @@ def clear(self, keep=None, clear=None, all=False):
-                p = tuple(p)  # to align with 'self.phis' format
-                lp = TruncateBasis  # = len(p[0][0])
 
-                raise ValueError("The method for the 'Cubic Splines' kernel has not yet been ported from development.")
+        ni_ids = []  # orders of basis functions used (where 0 is B1), per term
+        basis_n = []  # for future use when indexing 'm.fokl_basis'
+        for j in range(lv):  # for input variable in input variables
+            ni_ids.append(np.sort(np.unique(t[:, j][t[:, j] != -1])).tolist())
+            basis_n += ni_ids[j]
+        n_ids = np.sort(np.unique(basis_n))
 
-            return
-
-        elif self.kernel == self.kernels[1]:
-            p = []
-            for s in range(int(s_ids[-1] + 1)):  # for spline id in length of maximum spline id
-                if s in s_ids:
-                    p.append(self.phis[s])
-                else:
-                    p.append([])  # spline not used, but still need to maintain indexing
-            p = tuple(p)  # to align with 'self.phis' format
+        if self.kernel != self.kernels[1]:
+            raise ValueError("The method for the 'Cubic Splines' kernel has not yet been ported from development, nor "
+                             "is expected to be as the resulting symbolic expressions are too infeasible to solve. Use "
+                             "the 'to_pyomo' method with a FoKL model trained on the 'Bernoulli Polynomials' kernel.")
 
         if m is None:
             m = pyo.ConcreteModel()
 
-        m.j = pyo.Set(initialize=range(lv))  # index for FoKL input variable
-        m.y = pyo.Var(within=pyo.Reals)  # FoKL output
-        m.x = pyo.Var(m.j, within=pyo.Reals, bounds=[0, 1])  # FoKL input variables (per domain)
-
-        def fokl_as_pyomo(m):
-            """Convert FoKL model to Pyomo format assuming 'Bernoulli Polynomials' kernel."""
-            fokl = b[0]  # initialize expression with beta0 term
-            for k in range(lt):  # for term in terms
-                tk = t[k, :]
-                tk_mask = tk != -1  # recall -1 means ignore (i.e., input var with '-1 spline id' does not appear in term)
-                if any(tk_mask):
-                    term_k = b[k + 1]
-                    for j in range(lv):  # for variables that might be in term
-                        if tk_mask[j]:  # for variable in term
-                            n = int(tk[j])  # spline id, = sjs[j][k] if already referenced
-                            phix_sj = p[n][0] + sum(p[n][k] * (m.x[j] ** k) for k in range(1, len(p[n])))
-                            term_k *= phix_sj
-                else:
-                    term_k = 0
-                fokl += term_k  # append term to expression
-            return fokl
+        m.fokl_i = pyo.Set(initialize=range(draws))  # index for scenario (i.e., FoKL draw)
+        m.fokl_y = pyo.Var(m.fokl_i, within=pyo.Reals)  # FoKL output

```

Comment:
Index "y" over the set "fokl_scenarios"

Reply:
Does this mean "y" should be a single variable, instead of one var per scenario like it is now?

Reply:
If so, getting a "too few DOF" error from IPOPT so leaving as is for now

Reply:
Yes, this makes sense. I'll add a few more comments in a minute.